components/src/dynamo/trtllm/engine_monitor.py

```
@@ -0,0 +1,151 @@
+        """Shutdown the TRT-LLM engine on crash scenarios to free resources."""
+
+        def timeout_handler(signum, frame):
+            raise TimeoutError("TRT-LLM engine shutdown timed out")
+
+        if self.shutdown_timeout > 0:
+            signal.signal(signal.SIGALRM, timeout_handler)
+            signal.alarm(math.ceil(self.shutdown_timeout))
+
+        try:
+            self.engine.shutdown()
+        except Exception as exc:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)